docs: :memo: using `view` in CLI guide (#84)
seedcase-project/seedcase-flower: docs/guide/cli.qmd

description: "How to use Flower on the command-line."
order: 1
---

`seedcase-flower` is designed to provide a friendly user experience from
the command line. The CLI has commands for both viewing a building
documentation from a Data Package's metadata. The general pattern for
how to use seedcase-flower from the CLI is:

```{.bash filename="Terminal"}
seedcase-flower COMMAND [ARGS] [OPTIONS]
```

To get started, try calling the program without any commands, arguments,
or options by typing `seedcase-flower` and pressing enter. This is the
same as invoking the `--help` flag and will show the following text:

<!--TODO include output help text here when it is available -->

## Viewing Data Package metadata

To view the metadata of a Data Package directly in the terminal, you can
use the `view` command like so:

```{.bash filename="Terminal"}
seedcase-flower view
```

As you can see, the `view` command displays the metadata in an easy to
read format that is optimized for display directly in the terminal. By
default, `view` will look for a `datapackage.json` file in the current
directory, but you can specify a different path if desired:

```{.bash filename="Terminal"}

> seedcase-flower view another/directory/datapackage.json

Reviewer: Do we want to support point to the directory with a `datapackage.json` or should another path include the file name?

```

It is also possible to point flower to a Data Package that is available
online. To do this, you can provide the full URL to the remote
`datapackage.json`, e.g.

```{.bash filename="Terminal"}
seedcase-flower view https://raw.githubusercontent.com/seedcase-project/example-seed-beetle/refs/heads/main/datapackage.json
```

> For files that reside on GitHub, you can use the `gh:` prefix as

Reviewer 2: ```suggestion
For Data Packages on GitHub (like for the URL used above), you can use the `gh:` prefix as
```

a shortcut. For example, the long URL above can be shortened to the
following:

```{.bash filename="Terminal"}
seedcase-flower view gh:seedcase-project/example-seed-beetle
```

<!--TODO should we include output for all of these commands? collapsed by default -->

If you would like to customize the style of how the metadata is
displayed, you can use the `--style` flag. This option allows you to

> format the output with any of [the built-in styles](../design/interface/outputs.html#built-in-styles):

Reviewer 2: ```suggestion
format the output with any of [the built-in styles](/docs/design/interface/outputs.html#built-in-styles):
```

We generally try to use absolute paths in websites, as they are less likely to cause issues if this file is moved around. And makes it clearer where it is.

online. To do this, you can provide the full URL to the remote
`datapackage.json`, e.g.

```{.bash filename="Terminal"}
seedcase-flower view https://raw.githubusercontent.com/seedcase-project/example-seed-beetle/refs/heads/main/datapackage.json
```

For Data Packages on GitHub (like for the URL used above), you can use
the `gh:` prefix as a shortcut. For example, the long URL above can be
shortened to the following:

```{.bash filename="Terminal"}
seedcase-flower view gh:seedcase-project/example-seed-beetle
```

<!--TODO should we include output for all of these commands? collapsed by default -->

### Styling the output

If you would like to customize the style of how the metadata is
displayed, you can use the `--style` flag. This option allows you to
format the output with any of [the built-in
styles](../design/interface/outputs.html#built-in-styles):

<!-- TODO: point link above to guide for styles, or ref docs, not design -->

```{.bash filename="Terminal"}
seedcase-flower view --style terminal-simple gh:seedcase-project/example-seed-beetle
```

<!--TODO potentically change this depending on the name of this style when implemented -->

::: callout-important

> If you have set the `style` option in a configuration file read bpy

Reviewer 3: ```suggestion
If you have set the `style` option in a configuration file read by
```
